src/Picasso_FieldTypes.hpp

```diff
@@ -389,21 +429,36 @@ struct TensorViewWrapper
     // ordered as [i][j][k][dim0][dim1] if layout-right and
     // [dim0][dim1][k][j][i] if layout-left. Note the difference in
     // layout-left where the dim0 and dim1 dimensions are switched.
-    KOKKOS_FORCEINLINE_FUNCTION
-    LinearAlgebra::MatrixView<value_type, dim0, dim1>
+    template <int VR = view_rank>
+    KOKKOS_FORCEINLINE_FUNCTION std::enable_if_t<4 == VR, linear_algebra_type>
     operator()( const int i0, const int i1, const int i2 ) const
     {
-        return LinearAlgebra::MatrixView<value_type, dim0, dim1>(
-            &_v( i0, i1, i2, 0 ), dim1 * _v.stride( 3 ), _v.stride( 3 ) );
+        return linear_algebra_type( &_v( i0, i1, i2, 0 ), dim1 * _v.stride( 3 ),
```

Comment:
if tensor components are unrolled in the last dimension, `_v.stride(3) = dim0*dim1`? Then is correct way to access?

Reply:
The stride refers to the stride between field components in the Kokkos view. So here `_v.stride(3)` is either `1` if `Kokkos::LayoutLeft` or `NumI * NumJ * NumK` if `Kokkos::LayoutRight`

Reply:
The stride refers to the stride between the first component of each tensor in the Kokkos view. So here `_v.stride(3)` is either `1` if `Kokkos::LayoutLeft` or `NumI * NumJ * NumK` if `Kokkos::LayoutRight`

Reply:
So for 2D, (i=0,...,NumI), (j=0,...,NumJ) and tensor (M,N),  then `_v` is stored as first tensor component is stored at the grid NumI*NumJ and the second component of tensor is stored at the grid NumI*NumJ?